### ISSUE TITLE
Upgrade windows dependencies

### ISSUE THREAD

**Issue by kpcyrd** [carllerche/mio#870]
hey!

I've noticed the windows dependencies are outdated:

- winapi 0.2 was bumped to 0.3
- miow 0.2 was bumped to 0.3
- kernel32-sys 0.2 was been superseded by winapi 0.3

Getting this fixed would allow us to upload mio into debian, which is a significant step of uploading tokio.

Thanks!

**Comment by hawkw** [comment]
I believe updating the `winapi` version is a breaking change, which is why this hasn't been done yet (see https://github.com/carllerche/iovec/issues/16#issuecomment-405301640, #658). IIRC, updating the `winapi` dependency to 0.3 is planned for the 0.7 release of mio.

**Comment by steffengy** [comment]
It actually landed in the [0.7 branch](https://github.com/carllerche/mio/tree/v0.7.x) in january.
Since the [plan](https://github.com/carllerche/mio/issues/658#issuecomment-375712078) is/was to finish the remaining work for the 0.7 milestone to bundle up breaking changes,
I would - taking the past 6 months into account - not bank on this being released any time soon.

**Comment by carllerche** [comment]
As mentioned, it is a breaking change with far reaching impact.

The original plan has been to coordinate the release w/ `futures`, but I do not know when they plan to release now.

Either way, I will close this as the upgrade is being tracked.

**Comment by kpcyrd** [comment]
Is there a patch we could apply in debian to get mio 0.6 to work with winapi 0.3?

Given this was fixed in January and a release is not going to happen in the foreseeable future, did you consider splitting up the release until smaller releases? As somebody who pushed out a whole bunch of pull requests to update dependencies recently, adapting to 3 breaking changes over 3 releases or 3 breaking changes in a single release doesn't make much of a difference from a library consumer point of view. I've noticed that large breaking updates are actually more likely to cause outdated dependencies because it's too much effort at once.
